Build is failing with php-src master

```
...
mkdir:  cc -I. -I/tmp/pear/temp/igbinary -I/tmp/pear/temp/pear-build-defaultuserekdIGc/igbinary-3.2.13/include -I/tmp/pear/temp/pear-build-defaultuserekdIGc/igbinary-3.2.13/main -I/tmp/pear/temp/igbinary -I/usr/local/include/php -I/usr/local/include/php/main -I/usr/local/include/php/TSRM -I/usr/local/include/php/Zend -I/usr/local/include/php/ext -I/usr/local/include/php/ext/date/lib -DHAVE_CONFIG_H -g -O0 -D_GNU_SOURCE -g -O0 -O2 -Wall -Wpointer-arith -Wcast-align -Wwrite-strings -Wswitch -DZEND_COMPILE_DL_EXT=1 -c /tmp/pear/temp/igbinary/src/php7/igbinary.c -MMD -MF src/php7/igbinary.dep -MT src/php7/igbinary.lo  -fPIC -DPIC -o src/php7/.libs/igbinary.o
cannot create directory ‘src/php7/.libs’: File exists
 cc -I. -I/tmp/pear/temp/igbinary -I/tmp/pear/temp/pear-build-defaultuserekdIGc/igbinary-3.2.13/include -I/tmp/pear/temp/pear-build-defaultuserekdIGc/igbinary-3.2.13/main -I/tmp/pear/temp/igbinary -I/usr/local/include/php -I/usr/local/include/php/main -I/usr/local/include/php/TSRM -I/usr/local/include/php/Zend -I/usr/local/include/php/ext -I/usr/local/include/php/ext/date/lib -DHAVE_CONFIG_H -g -O0 -D_GNU_SOURCE -g -O0 -O2 -Wall -Wpointer-arith -Wcast-align -Wwrite-strings -Wswitch -DZEND_COMPILE_DL_EXT=1 -c /tmp/pear/temp/igbinary/src/php7/hash_si.c -MMD -MF src/php7/hash_si.dep -MT src/php7/hash_si.lo  -fPIC -DPIC -o src/php7/.libs/hash_si.o
/bin/sh /tmp/pear/temp/pear-build-defaultuserekdIGc/igbinary-3.2.13/libtool --mode=compile cc -I. -I/tmp/pear/temp/igbinary -I/tmp/pear/temp/pear-build-defaultuserekdIGc/igbinary-3.2.13/include -I/tmp/pear/temp/pear-build-defaultuserekdIGc/igbinary-3.2.13/main -I/tmp/pear/temp/igbinary -I/usr/local/include/php -I/usr/local/include/php/main -I/usr/local/include/php/TSRM -I/usr/local/include/php/Zend -I/usr/local/include/php/ext -I/usr/local/include/php/ext/date/lib  -DHAVE_CONFIG_H  -g -O0 -D_GNU_SOURCE   -g  -O0 -O2 -Wall -Wpointer-arith -Wcast-align -Wwrite-strings -Wswitch -DZEND_COMPILE_DL_EXT=1 -c /tmp/pear/temp/igbinary/src/php7/hash_si_ptr.c -o src/php7/hash_si_ptr.lo  -MMD -MF src/php7/hash_si_ptr.dep -MT src/php7/hash_si_ptr.lo
 cc -I. -I/tmp/pear/temp/igbinary -I/tmp/pear/temp/pear-build-defaultuserekdIGc/igbinary-3.2.13/include -I/tmp/pear/temp/pear-build-defaultuserekdIGc/igbinary-3.2.13/main -I/tmp/pear/temp/igbinary -I/usr/local/include/php -I/usr/local/include/php/main -I/usr/local/include/php/TSRM -I/usr/local/include/php/Zend -I/usr/local/include/php/ext -I/usr/local/include/php/ext/date/lib -DHAVE_CONFIG_H -g -O0 -D_GNU_SOURCE -g -O0 -O2 -Wall -Wpointer-arith -Wcast-align -Wwrite-strings -Wswitch -DZEND_COMPILE_DL_EXT=1 -c /tmp/pear/temp/igbinary/src/php7/hash_si_ptr.c -MMD -MF src/php7/hash_si_ptr.dep -MT src/php7/hash_si_ptr.lo  -fPIC -DPIC -o src/php7/.libs/hash_si_ptr.o
In file included from /tmp/pear/temp/igbinary/src/php7/hash_si_ptr.c:20:
/tmp/pear/temp/igbinary/src/php7/hash_ptr.h:34:9: error: unknown type name 'zend_uintptr_t'
   34 |         zend_uintptr_t key; /**< The key: The address of a pointer, casted to an int (won't be dereferenced). */
      |         ^~~~~~~~~~~~~~
/tmp/pear/temp/igbinary/src/php7/hash_ptr.h:102:64: error: unknown type name 'zend_uintptr_t'
  102 | size_t hash_si_ptr_find_or_insert(struct hash_si_ptr *h, const zend_uintptr_t key, uint32_t value);
      |                                                                ^~~~~~~~~~~~~~
/tmp/pear/temp/igbinary/src/php7/hash_si_ptr.c:25:47: error: unknown type name 'zend_uintptr_t'; did you mean 'tsrm_uintptr_t'?
   25 | inline static uint32_t inline_hash_of_address(zend_uintptr_t ptr) {
      |                                               ^~~~~~~~~~~~~~
      |                                               tsrm_uintptr_t
/tmp/pear/temp/igbinary/src/php7/hash_si_ptr.c: In function 'hash_si_ptr_rehash':
/tmp/pear/temp/igbinary/src/php7/hash_si_ptr.c:110:39: warning: implicit declaration of function 'inline_hash_of_address' [-Wimplicit-function-declaration]
  110 |                         uint32_t hv = inline_hash_of_address(old_data[i].key) & mask;
      |                                       ^~~~~~~~~~~~~~~~~~~~~~
/tmp/pear/temp/igbinary/src/php7/hash_si_ptr.c: At top level:
/tmp/pear/temp/igbinary/src/php7/hash_si_ptr.c:133:64: error: unknown type name 'zend_uintptr_t'
  133 | size_t hash_si_ptr_find_or_insert(struct hash_si_ptr *h, const zend_uintptr_t key, uint32_t value) {
      |                                                                ^~~~~~~~~~~~~~
make: *** [Makefile:216: src/php7/hash_si_ptr.lo] Error 1
make: *** Waiting for unfinished jobs....
ERROR: `make -j2' failed
```

caused probably by https://github.com/php/php-src/pull/10597

--- a/package.xml
+++ b/package.xml
@@ -31,10 +31,10 @@ memcached or similar memory based storages for serialized data.</description>
   <email>tandre@php.net</email>
   <active>yes</active>
  </lead>
- <date>2023-02-02</date>
+ <date>2023-02-26</date>
  <time>16:00:00</time>
  <version>
-  <release>3.2.13</release>
+  <release>3.2.14</release>
   <api>1.4.0</api>
  </version>
  <stability>
@@ -43,7 +43,8 @@ memcached or similar memory based storages for serialized data.</description>
  </stability>
  <license uri="https://github.com/igbinary/igbinary/blob/master/COPYING">BSD-3-Clause</license>
  <notes>
-* Speed up unserialization of typed properties by reducing hash table collisions when looking up property reference info.
+* Fix build error in PHP 8.3-dev
+* Fix test expectation errors in php 8.3-dev due to change to php's TypeErrors.
  </notes>
  <contents>
   <dir name="/">
@@ -239,6 +240,22 @@ memcached or similar memory based storages for serialized data.</description>
  <providesextension>igbinary</providesextension>
  <extsrcrelease />
  <changelog>
+  <release>
+   <date>2023-02-02</date>
+   <time>16:00:00</time>
+   <version>
+    <release>3.2.13</release>
+    <api>1.4.0</api>
+   </version>
+   <stability>
+    <release>stable</release>
+    <api>stable</api>
+   </stability>
+   <license uri="https://github.com/igbinary/igbinary/blob/master/COPYING">BSD-3-Clause</license>
+   <notes>
+* Speed up unserialization of typed properties by reducing hash table collisions when looking up property reference info.
+   </notes>
+  </release>
   <release>
    <date>2022-11-07</date>
    <time>16:00:00</time>

--- a/src/php7/hash_ptr.h
+++ b/src/php7/hash_ptr.h
@@ -5,7 +5,7 @@
   | Author: Oleg Grenrus <oleg.grenrus@dynamoid.com>                     |
   | See CREDITS for contributors                                         |
   | This defines hash_si_ptr.                                            |
-  | It is like hash_si, but the key is always a non-zero zend_uintptr_t  |
+  | It is like hash_si, but the key is always a non-zero uintptr_t       |
   +----------------------------------------------------------------------+
 */
 
@@ -31,19 +31,19 @@
  * @see hash_si_ptr
  */
 struct hash_si_ptr_pair {
-	zend_uintptr_t key; /**< The key: The address of a pointer, casted to an int (won't be dereferenced). */
-	uint32_t value;		/**< Value. */
+	uintptr_t key;  /**< The key: The address of a pointer, casted to an int (won't be dereferenced). */
+	uint32_t value; /**< Value. */
 };
 
 /** Hash-array.
- * Like c++ std::unordered_map<zend_uintptr_t, int32_t>, but does not allow HASH_PTR_KEY_INVALID as a key.
+ * Like c++ std::unordered_map<uintptr_t, int32_t>, but does not allow HASH_PTR_KEY_INVALID as a key.
  * Current implementation uses linear probing.
  * @author Oleg Grenrus <oleg.grenrus@dynamoid.com>
  */
 struct hash_si_ptr {
-	size_t size; 					/**< Allocated size of array. */
-	size_t used;					/**< Used size of array. */
-	struct hash_si_ptr_pair *data;		/**< Pointer to array or pairs of data. */
+	size_t size;                   /**< Allocated size of array. */
+	size_t used;                   /**< Used size of array. */
+	struct hash_si_ptr_pair *data; /**< Pointer to array or pairs of data. */
 };
 
 /** Inits hash_si_ptr structure.
@@ -67,7 +67,7 @@ void hash_si_ptr_deinit(struct hash_si_ptr *h);
  * @return 0 on success, 1 or 2 else.
  */
 /*
-int hash_si_ptr_insert (struct hash_si_ptr *h, const zend_uintptr_t key, uint32_t value);
+int hash_si_ptr_insert (struct hash_si_ptr *h, const uintptr_t key, uint32_t value);
 */
 
 /** Finds value from hash_si_ptr.
@@ -79,7 +79,7 @@ int hash_si_ptr_insert (struct hash_si_ptr *h, const zend_uintptr_t key, uint32_
  * @return 0 if found, 1 if not.
  */
 /*
-int hash_si_ptr_find (struct hash_si_ptr *h, const zend_uintptr_t key, uint32_t * value);
+int hash_si_ptr_find (struct hash_si_ptr *h, const uintptr_t key, uint32_t * value);
 */
 
 /** Returns size of hash_si_ptr.
@@ -99,7 +99,7 @@ static zend_always_inline size_t hash_si_ptr_size(struct hash_si_ptr *h) {
  *
  * @return SIZE_MAX or old, unmodified key
  */
-size_t hash_si_ptr_find_or_insert(struct hash_si_ptr *h, const zend_uintptr_t key, uint32_t value);
+size_t hash_si_ptr_find_or_insert(struct hash_si_ptr *h, const uintptr_t key, uint32_t value);
 
 /** Returns capacity of hash_si_ptr.
  * @param h Pointer to hash_si_ptr struct.

--- a/src/php7/hash_si_ptr.c
+++ b/src/php7/hash_si_ptr.c
@@ -22,7 +22,7 @@
 #include "igbinary_bswap.h"
 
 /* This assumes that pointers differ in low addresses rather than high addresses */
-inline static uint32_t inline_hash_of_address(zend_uintptr_t ptr) {
+inline static uint32_t inline_hash_of_address(uintptr_t ptr) {
 #if UINTPTR_MAX > UINT32_MAX
 	uint64_t hash = ptr;
 	hash *= 0x5e2d58d8b3bce8d9;
@@ -130,7 +130,7 @@ inline static void hash_si_ptr_rehash(struct hash_si_ptr *h) {
  * @param value - If the key does not exist, this is the value to associate with key
  * @return the old value, or SIZE_MAX if the key is brand new.
  */
-size_t hash_si_ptr_find_or_insert(struct hash_si_ptr *h, const zend_uintptr_t key, uint32_t value) {
+size_t hash_si_ptr_find_or_insert(struct hash_si_ptr *h, const uintptr_t key, uint32_t value) {
 	size_t size;
 	size_t mask;
 	uint32_t hv;

--- a/src/php7/igbinary.c
+++ b/src/php7/igbinary.c
@@ -1494,7 +1494,7 @@ inline static int igbinary_serialize_array(struct igbinary_serialize_data *igsd,
 /* TODO: Use different result codes for missing keys and errors */
 zend_always_inline static int igbinary_serialize_array_ref(struct igbinary_serialize_data *igsd, zval * const z, bool object) {
 	size_t t;
-	zend_uintptr_t key;  /* The address of the pointer to the zend_refcounted struct or other struct */
+	uintptr_t key;  /* The address of the pointer to the zend_refcounted struct or other struct */
 	static int INVALID_KEY;  /* Not used, but we take the pointer of this knowing other zvals won't share it*/
 
 	/* Similar to php_var_serialize_intern's first part, as well as php_add_var_hash, for printing R: (reference) or r:(object) */
@@ -1507,15 +1507,15 @@ zend_always_inline static int igbinary_serialize_array_ref(struct igbinary_seria
 	/* is_ref || IS_OBJECT implies it has a unique refcounted struct */
 	// NOTE: The original code would always use the same memory address - Z_COUNTED_P is the start of an object/array/reference
 // 	if (object && Z_TYPE_P(z) == IS_OBJECT) {
-// 		key = (zend_uintptr_t)Z_OBJ_P(z); /* expand object handle(uint32_t), cast to 32-bit/64-bit pointer */
+// 		key = (uintptr_t)Z_OBJ_P(z); /* expand object handle(uint32_t), cast to 32-bit/64-bit pointer */
 // 	} else if (is_ref) {
 // 		/* NOTE: PHP switched from `zval*` to `zval` for the values stored in HashTables.
 // 		 * If an array has two references to the same ZVAL, then those references will have different zvals.
 // 		 * We use Z_COUNTED_P(ref), which will be the same if (and only if) the references are the same. */
 // 		/* is_ref implies there is a unique reference counting pointer for the reference */
-// 		key = (zend_uintptr_t)Z_COUNTED_P(z);
+// 		key = (uintptr_t)Z_COUNTED_P(z);
 // 	} else if (EXPECTED(Z_TYPE_P(z) == IS_ARRAY)) {
-// 		key = (zend_uintptr_t)Z_ARR_P(z);
+// 		key = (uintptr_t)Z_ARR_P(z);
 // 	} else {
 // 		/* Nothing else is going to reference this when this is serialized, this isn't ref counted or an object, shouldn't be reached. */
 // 		/* Increment the reference id for the deserializer, give up. */
@@ -1528,9 +1528,9 @@ zend_always_inline static int igbinary_serialize_array_ref(struct igbinary_seria
 	/* because that's the way it was serialized in php5. */
 	/* Does this work with different forms of recursive arrays? */
 	if (igsd->references_id == 0 && !object) {
-		key = (zend_uintptr_t)&INVALID_KEY;
+		key = (uintptr_t)&INVALID_KEY;
 	} else {
-		key = (zend_uintptr_t)z->value.ptr;
+		key = (uintptr_t)z->value.ptr;
 	}
 
 	t = hash_si_ptr_find_or_insert(&igsd->references, key, igsd->references_id);
@@ -2845,7 +2845,7 @@ inline static int igbinary_unserialize_object_properties(struct igbinary_unseria
 					if (igsd->ref_props) {
 						/* Remove old entry from ref_props table, if it exists. */
 						zend_hash_index_del(
-							igsd->ref_props, ((zend_uintptr_t) prototype_value) >> ZEND_MM_ALIGNMENT_LOG2);
+							igsd->ref_props, ((uintptr_t) prototype_value) >> ZEND_MM_ALIGNMENT_LOG2);
 					}
 				}
 #endif
@@ -2927,7 +2927,7 @@ inline static int igbinary_unserialize_object_properties(struct igbinary_unseria
 					zend_hash_init(igsd->ref_props, 8, NULL, NULL, 0);
 				}
 				zend_hash_index_update_ptr(
-					igsd->ref_props, ((zend_uintptr_t) vp) >> ZEND_MM_ALIGNMENT_LOG2, info);
+					igsd->ref_props, ((uintptr_t) vp) >> ZEND_MM_ALIGNMENT_LOG2, info);
 			}
 		}
 #endif
@@ -3459,7 +3459,7 @@ static int igbinary_unserialize_zval(struct igbinary_unserialize_data *igsd, zva
 #if PHP_VERSION_ID >= 70400
 				zend_property_info *info = NULL;
 				if (igsd->ref_props) {
-					info = zend_hash_index_find_ptr(igsd->ref_props, ((zend_uintptr_t) z) >> ZEND_MM_ALIGNMENT_LOG2);
+					info = zend_hash_index_find_ptr(igsd->ref_props, ((uintptr_t) z) >> ZEND_MM_ALIGNMENT_LOG2);
 				}
 #endif
 				ZVAL_NEW_REF(z, z);

--- a/src/php7/igbinary.h
+++ b/src/php7/igbinary.h
@@ -18,7 +18,7 @@ struct zval;
 /** Binary protocol version of igbinary. */
 #define IGBINARY_FORMAT_VERSION 0x00000002
 
-#define PHP_IGBINARY_VERSION "3.2.13"
+#define PHP_IGBINARY_VERSION "3.2.14"
 
 /* Macros */
 

--- a/tests/__serialize_013_php8.phpt
+++ b/tests/__serialize_013_php8.phpt
@@ -6,6 +6,7 @@ __serialize() mechanism (013): Properties are still typed after unserialization 
 <?php
 declare(strict_types=1);
 
+/** @property mixed $std */
 #[AllowDynamicProperties]
 class Test {
     public int $i = 0;
@@ -34,7 +35,7 @@ $t2->s = null;
 try {
     $t2->s = false;
 } catch (Error $e) {
-    echo "s: " . $e->getMessage() . "\n";
+    echo "s: " . str_replace('false', 'bool', $e->getMessage()) . "\n";
 }
 $t2->s = 'other';
 try {
@@ -58,6 +59,7 @@ try {
     echo "a: " . $e->getMessage() . "\n";
 }
 var_dump($t2);
+?>
 --EXPECT--
 object(Test)#1 (5) {
   ["i"]=>

--- a/tests/__serialize_016_php8.phpt
+++ b/tests/__serialize_016_php8.phpt
@@ -9,8 +9,10 @@ if (PHP_VERSION_ID >= 90000) { echo "skip requires php < 9.0 when testing that t
 <?php
 declare(strict_types=1);
 
+// Asserts that this will emit a warning about dynamic properties due to the lack of #[AllowDynamicProperties] in PHP 8.2+, and suppresses the warning
 if (PHP_VERSION_ID >= 80200) { require_once __DIR__ . '/php82_suppress_dynamic_properties_warning.inc'; }
 
+/** @property mixed $std undeclared dynamic property */
 class Test {
     public int $i = 0;
     public ?string $s = 's';
@@ -39,7 +41,8 @@ $t2->s = null;
 try {
     $t2->s = false;
 } catch (Error $e) {
-    echo "s: " . $e->getMessage() . "\n";
+    // The error message is "Cannot assign false" in php 8.3+
+    echo "s: " . str_replace('false', 'bool', $e->getMessage()) . "\n";
 }
 $t2->s = 'other';
 try {
@@ -63,6 +66,7 @@ try {
     echo "a: " . $e->getMessage() . "\n";
 }
 var_dump($t2);
+?>
 --EXPECT--
 object(Test)#1 (5) {
   ["i"]=>

